src/ostorlab/runtimes/lite_local/agent_runtime.py

```diff
@@ -350,16 +350,21 @@ def create_agent_service(
         )
         caps = self.agent.caps or agent_definition.caps
 
-        service_name = (
-            agent_definition.service_name
-            or self.agent.container_image.split(":")[0].replace(".", "")
-            + "_"
-            + self.runtime_name
-        )
+        if agent_definition.service_name is not None:
+            # If the service name defined in the agent definition is longer than the max allowed service name
+            # we truncate it to avoid docker api error.
+            service_name = agent_definition.service_name[:MAX_SERVICE_NAME_LEN]
```

Comment:
Trucation of the provided service name is not expected, I would rather have an error at scan creation rather than this behavior.